comp/otelcol/otlp/components/processor/tagenrichmentprocessor/internal/metadata.go

```diff
@@ -0,0 +1,37 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package metadata - metadata for tag enrichment processor.
+package metadata
+
+import (
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/trace"
+)
+
+var (
+	// Type for tag enrichment processor.
+	Type = component.MustNewType("tagenrichment")
+)
+
+const (
+	// TracesStability - stability level for traces.
+	TracesStability = component.StabilityLevelAlpha
```

Comment:
Can you move it out of internal package ? We might need to use the types in our config provider.

Reply:
Done. I just moved metadata.go into the current package but I could see merging everything into  factory.go, lmk if you have a preference.